### PR TITLE
Middleware

### PR DESCRIPTION
**Pull request by sanukjoseph**

### CODE REVIEW

File: auth.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { DrizzleAdapter } from '@auth/drizzle-adapter';
+import AuthConfig from '~/auth.config';
+import NextAuth from 'next-auth';
+
+import { db } from './server/db';
+
+export const { handlers, auth, signIn, signOut } = NextAuth({
+  session: { strategy: 'jwt' },
+  adapter: DrizzleAdapter(db),
```
Comment (rajatsandeepsen) [inline auth.ts:9]: use `CustomPgDrizzleAdapter(db)` && `{ strategy: "database"}`

File: lib/constant.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+const publicRoutes: Routes = [
+  { path: '/design', access: 'public', roles: [] },
+  { path: '/home', access: 'public', roles: [] },
+  { path: '/events', access: 'public', roles: [] },
+  { path: '/events?workshops', access: 'public', roles: [] },
+  { path: '/events?competitons', access: 'public', roles: [] },
+  { path: '/events?events', access: 'public', roles: [] },
+  { path: '/asthra pass', access: 'public', roles: [] },
+  { path: '/events/[id]', access: 'public', roles: ['USER'] },
```
Comment (rajatsandeepsen) [inline lib/constant.ts:23]: one more route
`/events/edit/` only for `[studentCoordinator, Management and faculty]`

File: types/next-auth.d.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+export type ExtendedUser = DefaultSession['user'] &
+  ExtraData & {
+    id: string;
+  };
+
+declare module 'next-auth' {
+  interface Session {
+    user: ExtendedUser;
+  }
+}
```
Comment (rajatsandeepsen) [inline types/next-auth.d.ts:14]: Extending user data doesn't work in jwt (i think)

File: lib/constant.ts (hunk 2 above)
Comment (sanukjoseph) [reply lib/constant.ts:23]: Working on it..

**Pull request merged**